azure-pipelines.yml

```diff
@@ -125,15 +160,24 @@ stages:
           verbosityPack: 'Detailed'
           publishPackageMetadata: true
 
-      # Code signing is temporarily disabled until new certificate is properly set up.
-      # https://github.com/SonarSource/sonar-dotnet/issues/8230
-      - task: NuGetCommand@2
-        enabled: false
+      - task: PowerShell@2
         displayName: "Sign NuGet packages"
-        condition: eq(variables.isReleaseBranch, 'True')
-        inputs:
-          command: 'custom'
-          arguments: 'sign $(Build.ArtifactStagingDirectory)\packages\*.nupkg -CertificatePath $(pfx.secureFilePath) -CertificatePassword $(pfxPassword2021-2023) -Timestamper http://sha256timestamp.ws.symantec.com/sha256/timestamp'
+        # ToDo: add back the condition
+        # condition: eq(variables.isReleaseBranch, 'True')
+        env:
+          PACKAGES_PATH: '$(Build.ArtifactStagingDirectory)\packages\*.nupkg'
+          SIGNTOOL_PATH: $(SignToolPath)
+          SM_CLIENT_CERT_FILE: $(SM_CLIENT_CERT.secureFilePath)
+          SM_CLIENT_CERT_PASSWORD: $(SM_CLIENT_CERT_PASSWORD)
+          SM_API_KEY: $(SM_API_KEY)
+          SM_CERT: $(SM_CERT)
+          SM_CERT_FP: $(SM_CERT_FP)
+          SM_HOST: $(SM_HOST)
+          SM_KP: $(SM_KP)
+        inputs:
+          targetType: 'inline'
+          script: |
+            nuget sign "$env:PACKAGES_PATH" -Overwrite -HashAlgorithm SHA256 -CertificateFingerprint $env:SM_CERT_FP -Timestamper http://timestamp.digicert.com -TimestampHashAlgorithm SHA256
```

Comment:
There is no need to copy `$env:SM_CERT_FP` to the env and access it here. You can use the variable $(SM_CERT_FP) directly here.